### ISSUE TITLE
MD033: Consider not reporting closing tags and HTML comments separately

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#278]
## Summary

MD033 reports significantly more violations than other linters because we report both opening AND closing tags, plus HTML comments.

## Comparison

| Linter | MD033 Violations |
|--------|------------------|
| mdbook-lint | 1096 |
| rumdl | 801 |
| markdownlint | 800 |

### 1. We report closing tags separately
```
<span class="filename">  -> violation
</span>                  -> another violation
```
rumdl only reports the opening tag, which is sufficient to flag the HTML usage.

### 2. We report HTML comments
```
<!-- ignore -->  -> violation
```
rumdl does not flag HTML comments, which are often intentional (for mdBook directives, editor hints, etc.)

## Proposed Changes

1. **Option A**: Only report opening tags, skip closing tags
2. **Option B**: Add config option to allow HTML comments
3. **Option C**: Add config option `allowed_elements` that defaults to allowing comments

## Related

Issue #263 - Linter comparison
